geomstats/geometry/quotient_manifold.py

```diff
@@ -0,0 +1,480 @@
+    """
+
+    def __init__(self, total_space, base=None, group=None, group_action=None,
+                 dim=None, **kwargs):
+
+        if dim is None:
+            if base is not None:
+                dim = base.dim
+            elif group is not None:
+                dim = total_space.dim - group.dim
+            else:
+                raise ValueError('Either the base manifold, '
+                                 'its dimension, or the group acting on the '
+                                 'total space must be provided.')
+
+        super(FiberBundle, self).__init__(dim=dim, **kwargs)
+
+        self.base = base
+        self.total_space = total_space
+        self.group = group
+
+        if group_action is None and group is not None:
+            group_action = group.compose
+        self.group_action = group_action
+
+    def belongs(self, point, atol=1e-6):
+        """Evaluate if a point belongs to the base manifold.
+
+        Evaluate if a point belongs to the base manifold when it is given,
+        otherwise to the total space.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., dim]
+            Point to evaluate.
+        atol : float
+            Absolute tolerance.
+            Optional, default: 1e-6.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean evaluating if point belongs to the manifold.
+        """
+        if self.base is not None:
+            return self.base.belongs(point, atol=atol)
+        return self.total_space.belongs(point, atol=atol)
+
+    @staticmethod
+    def submersion(point):
+        """Project a point to base manifold.
+
+        This is the projection of the fiber bundle, defined on the total
+        space, with values in the base manifold. This map is surjective.
+        By default, the base manifold  is not explicit but is identified with a
+        local section of the fiber bundle, so the submersion is the identity
+        map.
+
+        Parameters
+        ----------
+        point: array-like, shape=[..., {ambient_dim, [n, n]}]
+            Point of the total space.
+
+        Returns
+        -------
+        projection: array-like, shape=[..., {dim, [n, n]}]
+            Point of the base manifold.
+        """
+        return point
+
+    @staticmethod
+    def lift(point):
+        """Lift a point to total space.
+
+        This is a section of the fiber bundle, defined on the base manifold,
+        with values in the total space. It means that submersion applied after
+        lift results in the identity map. By default, the base manifold
+        is not explicit but is identified with a section of the fiber bundle,
+        so the lift is the identity map.
+
+        Parameters
+        ----------
+        point: array-like, shape=[..., {dim, [n, n]}]
+            Point of the base manifold.
+
+        Returns
+        -------
+        lift: array-like, shape=[..., {ambient_dim, [n, n]}]
```

Comment:
NIT: maybe not use "lift", since there is no variable called lift here.

Reply:
don't we do this very often?

Reply:
Yes we do do it... it's confusing though.